### PR TITLE
MULE-19622: add munit redirect tests

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

File: src/test/munit/redirect/redirect-test-case.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
+      xmlns:munit-tools="http://www.mulesoft.org/schema/mule/munit-tools"
+      xsi:schemaLocation="
+               http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+               http://www.mulesoft.org/schema/mule/tls http://www.mulesoft.org/schema/mule/tls/current/mule-tls.xsd
+               http://www.mulesoft.org/schema/mule/munit http://www.mulesoft.org/schema/mule/munit/current/mule-munit.xsd
+		       http://www.mulesoft.org/schema/mule/munit-tools http://www.mulesoft.org/schema/mule/munit-tools/current/mule-munit-tools.xsd
+               http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd">
+
+    <munit:config name="redirect-test-case" minMuleVersion="4.3.0"/>
```
Comment (elrodro83) [inline src/test/munit/redirect/redirect-test-case.xml:15]: why the minMuleVersion?

File: src/test/munit/redirect/redirect-test-case.xml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    <http:request-config name="followRedirects" followRedirects="true">
+        <http:request-connection host="localhost" port="${dynamic.port}"/>
+    </http:request-config>
+
+    <http:request-config name="noRedirect" followRedirects="false">
+        <http:request-connection host="localhost" port="${dynamic.port}"/>
+    </http:request-config>
+
+    <munit:test name="moved-test" description="Test">
```
Comment (elrodro83) [inline src/test/munit/redirect/redirect-test-case.xml:31]: add meaningful descripotion (applies the other tests below)

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Comment by jcgrethe** [comment]
--validate

File: src/test/munit/redirect/redirect-test-case.xml [hunk 3]
```diff
@@ -5,14 +5,16 @@
       xmlns:tls="http://www.mulesoft.org/schema/mule/tls"
       xmlns:munit="http://www.mulesoft.org/schema/mule/munit"
       xmlns:munit-tools="http://www.mulesoft.org/schema/mule/munit-tools"
+      xmlns:java="http://www.mulesoft.org/schema/mule/java"
```
Comment (elrodro83) [inline src/test/munit/redirect/redirect-test-case.xml:8]: why did you add this?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/test/munit/redirect/redirect-test-case.xml (hunk 3 above)
Comment (jcgrethe) [reply src/test/munit/redirect/redirect-test-case.xml:8]: I wanted to check an exception and I didn't remember to remove it.

**Review by elrodro83 — APPROVED** [review]
